bundles/org.openhab.binding.awattar/src/main/java/org/openhab/binding/awattar/internal/handler/AwattarBridgeHandler.java

```diff
@@ -53,34 +54,30 @@
  * The {@link AwattarBridgeHandler} is responsible for retrieving data from the aWATTar API.
  *
  * The API provides hourly prices for the current day and, starting from 14:00, hourly prices for the next day.
- * Check the documentation at https://www.awattar.de/services/api
+ * Check the documentation at <a href="https://www.awattar.de/services/api" />
  *
  *
  *
  * @author Wolfgang Klimt - Initial contribution
  */
 @NonNullByDefault
 public class AwattarBridgeHandler extends BaseBridgeHandler {
+    private static final int DATA_REFRESH_INTERVAL = 60;
+
     private final Logger logger = LoggerFactory.getLogger(AwattarBridgeHandler.class);
     private final HttpClient httpClient;
-    @Nullable
-    private ScheduledFuture<?> dataRefresher;
+    private @Nullable ScheduledFuture<?> dataRefresher;
 
     private static final String URLDE = "https://api.awattar.de/v1/marketdata";
     private static final String URLAT = "https://api.awattar.at/v1/marketdata";
     private String url;
 
     // This cache stores price data for up to two days
-    @Nullable
-    private SortedMap<Long, AwattarPrice> priceMap;
-    private final int dataRefreshInterval = 60;
+    private @Nullable SortedSet<AwattarPrice> prices;
```

Comment:
I am not sure regarding the use of a Set instead of a Map here. I started with a set here also, with the price as sortKey. The problem with this was that sometimes there are different hours on a day with exactly the same price, and in that cases my Set lost one entry. This is why I then switched to a map.

I did not completely understand the logic you use switched to, but your awattarPrice comparator still only compares the prices. As you are about to write tests (which I was too lazy for, sorry), please double-check this with a test case containing the same price for different hours to ensure that the code works correctly in this cases also.